packages/derived-wallet-solana/src/signAptosTransaction.ts

```diff
@@ -39,38 +37,67 @@ export async function signAptosTransactionWithSolana(input: SignAptosTransaction
   const signingMessage = generateSigningMessageForTransaction(rawTransaction);
   const signingMessageDigest = hashValues([signingMessage]);
 
-  const siwsInput = createSiwsEnvelopeForAptosTransaction({
-    solanaPublicKey,
-    rawTransaction,
-    signingMessageDigest,
-    domain
-  });
 
-  const response = await wrapSolanaUserResponse(solanaWallet.signIn!(siwsInput));
-  return mapUserResponse(response, (output): AccountAuthenticator => {
-    if (output.signatureType && output.signatureType !== 'ed25519') {
-      throw new Error('Unsupported signature type');
-    }
+  // Prioritize SIWS if available
+  if(solanaWallet.signIn){
+    const siwsInput = createSiwsEnvelopeForAptosTransaction({
+      solanaPublicKey,
+      rawTransaction,
+      signingMessageDigest,
+      domain
+    });
+
+    const response = await wrapSolanaUserResponse(solanaWallet.signIn!(siwsInput));
+    return mapUserResponse(response, (output): AccountAuthenticator => {
+      if (output.signatureType && output.signatureType !== 'ed25519') {
+        throw new Error('Unsupported signature type');
+      }
+  
+      // The wallet might change some of the fields in the SIWS input, so we
+      // might need to include the finalized input in the signature.
+      // For now, we can assume the input is unchanged.
+      const signature = new Ed25519Signature(output.signature);
+  
+      return createAccountAuthenticatorForSolanaTransaction(signature, solanaPublicKey, domain, authenticationFunction, signingMessageDigest);
+    });
+  }else if(solanaWallet.signMessage){
+    // Fallback to signMessage if SIWS is not available
+    const signMessageInput = createSolanaSignMessageStatementForAptosTransaction({accountAddress: solanaPublicKey.toBase58(), signingMessageDigest, rawTransaction});
+    const signMessageInputBytes = new TextEncoder().encode(signMessageInput);
```

Comment:
Add the following above
```typescript
import { createSignInMessage } from '@solana/wallet-standard-util';
```

Then replace with
```suggestion
    const signMessageInputBytes = createSignInMessage(siwsInput);
```

which btw internally looks like this:
```typescript
export function createSignInMessage(input: SolanaSignInInputWithRequiredFields): Uint8Array {
    const text = createSignInMessageText(input);
    return new TextEncoder().encode(text);
}
```

Reply:
then we can remove `createSolanaSignMessageStatementForAptosTransaction` altogether I think?

Reply:
nice!